File: app/src/main/java/bj/rxjavaexperimentation/model/release/Company.java

package bj.rxjavaexperimentation.model.release;

import com.google.gson.annotations.Expose;
import com.google.gson.annotations.SerializedName;

public class Company {

    @SerializedName("catno")
    @Expose
    private String catno;
    @SerializedName("entity_type")
    @Expose
    private String entityType;
    @SerializedName("entity_type_name")
    @Expose
    private String entityTypeName;
    @SerializedName("id")
    @Expose
    private Integer id;
    @SerializedName("name")
    @Expose
    private String name;
    @SerializedName("resource_url")
    @Expose
    private String resourceUrl;

    public String getCatno() {
        return catno;
    }

    public void setCatno(String catno) {
        this.catno = catno;
    }

    public String getEntityType() {
        return entityType;
    }

    public void setEntityType(String entityType) {
        this.entityType = entityType;
    }

    public String getEntityTypeName() {
        return entityTypeName;
    }

    public void setEntityTypeName(String entityTypeName) {
        this.entityTypeName = entityTypeName;
    }

    public Integer getId() {
        return id;
    }

    public void setId(Integer id) {
        this.id = id;
    }

    public String getName() {
        return name;
    }

    public void setName(String name) {
        this.name = name;
    }

    public String getResourceUrl() {
        return resourceUrl;
    }

    public void setResourceUrl(String resourceUrl) {
        this.resourceUrl = resourceUrl;
    }

}

File: app/src/main/java/bj/vinylbrowser/model/release/Company.kt
package bj.vinylbrowser.model.release

import android.os.Parcel
import android.os.Parcelable
import com.google.gson.annotations.SerializedName

/**
 * Created by Josh Laird on 19/05/2017.
 */
data class Company(val catno: String,
                   @SerializedName("entity_type") val entityType: String,
                   @SerializedName("entity_type_name") val entityTypeName: String,
                   val id: Int,
                   val name: String,
                   @SerializedName("resource_url") val resourceUrl: String) : Parcelable {
    override fun writeToParcel(dest: Parcel, flags: Int) {
        dest.writeString(catno)
        dest.writeString(entityType)
        dest.writeString(entityTypeName)
        dest.writeInt(id)
        dest.writeString(name)
        dest.writeString(resourceUrl)
    }

    override fun describeContents(): Int {
        return 0
    }
}